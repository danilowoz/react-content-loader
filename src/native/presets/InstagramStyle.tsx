import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader, { Circle, Rect } from '../ContentLoader'

const ReactContentLoaderInstagram: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 400 460" width={400} height={460} {...props}>
    <Circle cx="31" cy="31" r="15" />
    <Rect x="58" y="18" rx="2" ry="2" width="140" height="10" />
    <Rect x="58" y="34" rx="2" ry="2" width="140" height="10" />
    <Rect x="0" y="60" rx="2" ry="2" width="400" height="400" />
  </ContentLoader>
)

export default ReactContentLoaderInstagram
