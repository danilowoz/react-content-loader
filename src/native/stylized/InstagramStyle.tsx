import * as React from 'react'

import Holder, { Rect, Circle } from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderInstagram: React.FC<IContentLoaderProps> = props => (
  <Holder {...props} height={480}>
    <Circle cx="30" cy="30" r="30" />

    <Rect x="75" y="13" rx="4" ry="4" width="100" height="13" />
    <Rect x="75" y="37" rx="4" ry="4" width="50" height="8" />
    <Rect x="0" y="70" rx="5" ry="5" width="400" height="400" />
  </Holder>
)

export default ReactContentLoaderInstagram
