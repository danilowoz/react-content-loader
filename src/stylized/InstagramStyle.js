// @flow
import * as React from "react"
import ContentLoader from "../index"
import type { WrapProps } from "../Wrap"

const InstagramStyle = (props: WrapProps): React.Element<*> => (
  <ContentLoader {...props} height={480}>
    <circle cx="30" cy="30" r="30" />

    <rect x="75" y="13" rx="4" ry="4" width="100" height="13" />
    <rect x="75" y="37" rx="4" ry="4" width="50" height="8" />
    <rect x="0" y="70" rx="5" ry="5" width="400" height="400" />
  </ContentLoader>
)

export default InstagramStyle
