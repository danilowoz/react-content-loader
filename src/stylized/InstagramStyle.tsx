import * as React from 'react'

import Holder from '../Holder'
import { IContentLoaderProps } from '../'

export default (props: IContentLoaderProps) => (
  <Holder {...props} height={480}>
    <circle cx="30" cy="30" r="30" />

    <rect x="75" y="13" rx="4" ry="4" width="100" height="13" />
    <rect x="75" y="37" rx="4" ry="4" width="50" height="8" />
    <rect x="0" y="70" rx="5" ry="5" width="400" height="400" />
  </Holder>
)
