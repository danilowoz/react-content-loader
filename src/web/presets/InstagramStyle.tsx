import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader from '../ContentLoader'

const ReactContentLoaderInstagram: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 400 460" {...props}>
    <circle cx="31" cy="31" r="15" />
    <rect x="58" y="18" rx="2" ry="2" width="140" height="10" />
    <rect x="58" y="34" rx="2" ry="2" width="140" height="10" />
    <rect x="0" y="60" rx="2" ry="2" width="400" height="400" />
  </ContentLoader>
)

export default ReactContentLoaderInstagram
