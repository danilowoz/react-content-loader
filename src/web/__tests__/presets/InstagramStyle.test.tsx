import * as React from 'react'
import * as renderer from 'react-test-renderer'

import InstagramStyle from '../../presets/InstagramStyle'

describe('InstagramStyle', () => {
  const wrapper = renderer.create(
    <InstagramStyle uniqueKey="InstagramStyle" speed={20} />
  )

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
    expect(wrapper.root.props.uniqueKey).toEqual('InstagramStyle')
  })
})
