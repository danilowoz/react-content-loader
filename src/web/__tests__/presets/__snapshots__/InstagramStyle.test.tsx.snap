// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstagramStyle renders correctly 1`] = `
<svg
  aria-labelledby="InstagramStyle-aria"
  id={null}
  role="img"
  style={Object {}}
  viewBox="0 0 400 460"
>
  <title
    id="InstagramStyle-aria"
  >
    Loading...
  </title>
  <rect
    clipPath="url(#InstagramStyle-diff)"
    height="100%"
    role="presentation"
    style={
      Object {
        "fill": "url(#InstagramStyle-animated-diff)",
      }
    }
    width="100%"
    x="0"
    y="0"
  />
  <defs
    role="presentation"
  >
    <clipPath
      id="InstagramStyle-diff"
    >
      <circle
        cx="31"
        cy="31"
        r="15"
      />
      <rect
        height="10"
        rx="2"
        ry="2"
        width="140"
        x="58"
        y="18"
      />
      <rect
        height="10"
        rx="2"
        ry="2"
        width="140"
        x="58"
        y="34"
      />
      <rect
        height="400"
        rx="2"
        ry="2"
        width="400"
        x="0"
        y="60"
      />
    </clipPath>
    <linearGradient
      id="InstagramStyle-animated-diff"
    >
      <stop
        offset="0%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-2; -2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#eee"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-1; -1; 2"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="0; 0; 3"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
